src/dbus/server/introspect.xml

```diff
@@ -10,14 +10,14 @@
         struct {
           uint64 ext_address
           uint16 panid
-          uint16 channel
-          uint16 rssi
-          uint8 lqi
+          uint8  channel
+          int8   rssi
```

Comment:
d-bus doesn't support signed byte. Should we use int16 instead?
```suggestion
          int16  rssi
```

Reply:
I can see three different paths for changing it:

1. Change the type in `ActiveScanResult`. `DBusMessageEncode` eventually passes the data its trying to use [through a template](https://github.com/openthread/ot-br-posix/blob/0253f26ce53b1db6a4f395c3273fde353fd3901b/src/dbus/common/dbus_message_helper.hpp#L389-L393) to get the type. The easiest fix would be to change the type to force the encoder to encode it as an `int16`. This would make the types inconsistent between [`otActiveScanResult`](https://github.com/openthread/openthread/blob/6a21b5f9ae29dd165ea2a359943264910048957e/include/openthread/link.h#L376-L398) and `ActiveScanResult` though.
2.  Define a specific trait template for the `ActiveScanResult` struct that would end up encoding the value as a `int16`. It would keep the data structures consistent, but introduce complexity. I could also see it causing confusion.
3. Instead of an entire struct, define a specific trait for this one value (newtype pattern). I think it has the same problems as 2 though.

If the type is left as `int8` in the return, it could also be documented that it should be cast as a signed value on the client end.

Let me know what you think. I'm indifferent on the outcome as I can make the platform I'm supporting work in whichever way is deemed appropriate. I just need it to not crash. 😄

Reply:
Fixed!